public/lesswrong-editor/src/ckeditor5-dialogue-comments/dialogue-comment-box.js

```diff
@@ -296,19 +329,47 @@ class SubmitDialogueMessageCommand extends Command {
 
                 if (dialogueMessageInput.is('element')) {
                     writer.append(dialogueMessage, root);
+
+                    const contentDiv = writer.createElement('dialogueMessageContent');
+
                     const inputContents = Array.from(dialogueMessageInput.getChildren());
                     if (inputContents.length === 0) {
-                        writer.appendElement('paragraph', dialogueMessage);
+                        const paragraph = writer.createElement('paragraph');
+                        writer.append(paragraph, contentDiv);
+                        writer.append(paragraph, dialogueMessage);
                     }
 
-                    Array.from(dialogueMessageInput.getChildren()).forEach(userInput => {
-                        writer.append(userInput, dialogueMessage);
+                    inputContents.forEach(userInput => {
+                        const paragraph = writer.createElement('paragraph');
+                        let userInputText = userInput;
+                        let iterations = 0;
+                        while (!userInputText.is('text') && iterations < 5) {
+                            const child = userInputText.getChild(0);
+                            if (child) {
+                                userInputText = child;
+                            } else {
+                                break;
+                            }
+                            iterations++;
+                        }
+
+                        // at this point, userInput should be Text, or something's gone wrong
+                        if (userInputText.is('text')) {
+                            writer.insertText(userInputText.data, paragraph);
+                        }
+                        writer.append(paragraph, contentDiv);
+                        writer.remove(userInput);
                     });
+
+                    writer.append(contentDiv, dialogueMessage);
+
                     // After we are done moving, add a new paragraph to dialogueMessageInput, so it's not empty
                     writer.appendElement('paragraph', dialogueMessageInput);
                 } else {
-                    writer.insertText(dialogueMessageInput.data, dialogueMessage);
-                    writer.append(dialogueMessage, root);
+                    // const messageChildren = Array.from(dialogueMessage.getChildren());
+                    // writer.insertText(dialogueMessageInput.data, messageChildren[0]);
+                    console.log(`else block of dialogueMessageInput.is('element')`, { dialogueMessage });
+                    // writer.append(dialogueMessage, root);
```

Comment:
I don't think this ever gets hit but we should clean it up (maybe log an error, or something)